### PR TITLE
Do not generate keydown event for empty modifier flags

### PR DESCRIPTION
**Pull request by knopp**
Fixes https://github.com/flutter/flutter/issues/87339

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponder.mm [hunk 1]
```diff
@@ -39,6 +39,8 @@ - (nonnull instancetype)initWithChannel:(nonnull FlutterBasicMessageChannel*)cha
 }
 
 - (void)handleEvent:(NSEvent*)event callback:(FlutterAsyncKeyCallback)callback {
+  // Remove the 0x100 bit set by Cocoa when no modifiers are pressed.
```
Comment (dkwingsmt) [inline shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponder.mm:42]:
```suggestion
  // Remove the modifier bits that Flutter is not interested in. 
```

**Review by dkwingsmt — APPROVED** [review]
LGTM with doc suggestion

File: shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponder.mm (hunk 1 above)
Comment (knopp) [reply shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponder.mm]: Done.

File: shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponderUnittests.mm [hunk 2]
```diff
@@ -52,9 +52,19 @@
         callback(keyMessage);
       }));
 
-  // Key down
   FlutterChannelKeyResponder* responder =
       [[FlutterChannelKeyResponder alloc] initWithChannel:mockKeyEventChannel];
+
+  // Initial empty modifiers. This can happen when user opens window while modifier key is pressed
+  // and then releases the modifier. Shouldn't result in an event being sent.
```
Comment (dkwingsmt) [inline shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponderUnittests.mm:59]:
```suggestion
  // and then releases the modifier. Shouldn't result in an event being sent.
  // Regression test for https://github.com/flutter/flutter/issues/87339.
```

Comment (knopp) [reply shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponderUnittests.mm:59]: Done.

**Pull request merged**